core/src/main/java/com/graphhopper/routing/util/MotorcycleFlagEncoder.java

```diff
@@ -350,19 +365,63 @@ public void applyWayTags( OSMWay way, EdgeIteratorState edge )
     {
         double speed = this.getSpeed(edge.getFlags());
         double roadDistance = edge.getDistance();
-        double beelineDistance = getBeelineDistance(way);
+        double beelineDistance = getBeelineDistance(edge);
+        PointList pointList = edge.fetchWayGeometry(3);
```

Comment:
Yes sure, I thought about that as well, but thought the method call would not be expensive.